File: src/api/subsonic/types/baseArtist.ts
import * as v from 'valibot';

export const BaseArtist = v.object({
  albumCount: v.number(),
  coverArt: v.string(),
  id: v.string(),
  musicBrainzId: v.optional(v.string()),
  name: v.string(),
  starred: v.optional(v.string()),
  userRating: v.optional(v.number()),
});

export type BaseArtist = v.InferInput<typeof BaseArtist>;

File: src/api/subsonic/baseArtist.ml
open Ppx_deriving_json_runtime.Primitives

type[@deriving json] t = {
  albumCount : int;
  coverArt : string;
  id : string;
  musicBrainzId : string option; [@json.option]
  name : string;
  starred : string option; [@json.option]
  userRating : int option; [@json.option]
}
